x-pack/plugins/synthetics/server/legacy_uptime/lib/saved_objects/migrations/monitors/8.8.0.ts

```diff
@@ -118,3 +126,73 @@ const omitZipUrlFields = (fields: BrowserFields) => {
 
   return formatSecrets(validationResult.decodedMonitor);
 };
+
+const updateThrottlingFields = (
+  doc: SavedObjectUnsanitizedDoc<
+    SyntheticsMonitorWithSecrets &
+      Partial<{
+        [LegacyConfigKey.THROTTLING_CONFIG]: string;
+        [LegacyConfigKey.IS_THROTTLING_ENABLED]: boolean;
+        [LegacyConfigKey.DOWNLOAD_SPEED]: string;
+        [LegacyConfigKey.UPLOAD_SPEED]: string;
+        [LegacyConfigKey.LATENCY]: string;
+      }>
+  >
+) => {
+  const { attributes } = doc;
+
+  const migrated = {
+    ...doc,
+    attributes: {
+      ...doc.attributes,
+      [ConfigKey.CONFIG_HASH]: '',
+    },
+  };
+
+  const isThrottlingEnabled = attributes[LegacyConfigKey.IS_THROTTLING_ENABLED];
+  if (isThrottlingEnabled) {
+    const download = String(attributes[LegacyConfigKey.DOWNLOAD_SPEED])!;
+    const upload = String(attributes[LegacyConfigKey.UPLOAD_SPEED])!;
+    const latency = String(attributes[LegacyConfigKey.LATENCY])!;
+
+    const isDefault = isDefaultThrottlingConfig(download, upload, latency);
+
+    const newThrottlingConfig: ThrottlingConfig = {
+      value: {
+        download,
+        upload,
+        latency,
+      },
+      id: isDefault ? PROFILE_VALUES_ENUM.DEFAULT : 'custom',
+      label: isDefault ? 'Default' : `Custom`,
+    };
+
+    // @ts-ignore
+    migrated.attributes[ConfigKey.THROTTLING_CONFIG] = newThrottlingConfig;
+  } else {
+    // @ts-ignore
+    migrated.attributes[ConfigKey.THROTTLING_CONFIG] =
+      PROFILES_MAP[PROFILE_VALUES_ENUM.NO_THROTTLING];
+  }
+
+  // filter out legacy throttling fields
+  return {
+    ...migrated,
+    attributes: omit(migrated.attributes, [
```

Comment:
When I was removing zip url, I went ahead and ran the config through the `validate` function, which strips off any fields that are not part of the current type definition. You could consider doing that, but this is fine too.